Library/Homebrew/os/mac/linkage_checker.rb

```diff
@@ -62,13 +63,31 @@ def check_undeclared_deps
       formula.build.without?(dep)
     end
     declared_deps = formula.deps.reject { |dep| filter_out.call(dep) }.map(&:name)
+    recursive_deps = keg.to_formula.runtime_dependencies.map { |dep| dep.to_formula.full_name }
     declared_dep_names = declared_deps.map { |dep| dep.split("/").last }
-    undeclared_deps = @brewed_dylibs.keys.reject do |full_name|
+    indirect_deps = []
+    undeclared_deps = []
+    @brewed_dylibs.each_key do |full_name|
       name = full_name.split("/").last
-      next true if name == formula.name
-      declared_dep_names.include?(name)
+      next if name == formula.name
+      if recursive_deps.include?(name)
+        indirect_deps << full_name unless declared_dep_names.include?(name)
+      else
+        undeclared_deps << full_name
+      end
+    end
+    sort_by_formula_full_name!(undeclared_deps)
+    sort_by_formula_full_name!(indirect_deps)
```

Comment:
would probably do this sort first just for aesthetics since they're mentioned in indirect, then undeclared, order above not the other way around

Reply:
Makes sense. Updated.